### PR TITLE
fix(tui): expose task tail errors and polling options

### LINKED ISSUE #1824: Centralize task-output tail polling across workbench surfaces
Verified on `main` at `065318d8f`.

## Problem

ShellSurface, AgentSurface, and TestSurface separately implement task-output path lookup, tail reads, polling, cancellation, error retention, and status handling. The copies already differ in byte limits and output retention.

## Suggested fix

Create `useTaskOutputTail` with explicit task ID, status, byte limit, and poll interval inputs. The hook should preserve content for the same task, reset on task changes, abort stale reads, stop polling in terminal states after one final read, and return an observable error without discarding good content.

Keep surface-specific parsing and rendering outside the hook.

## Acceptance criteria

- All three surfaces use the shared hook.
- Polling and final-read behavior are identical.
- Tests cover task switches, status transitions, transient read errors, unmount, and late promises.
- Byte-limit differences are named caller inputs rather than copied constants.

### PR DESCRIPTION
**Pull request by 7etsuo**
## Changes

- Complete the shared task-tail hook introduced in #2269. Task ID, status, byte limit, and polling interval are named caller inputs.
- Return read errors alongside retained output. Shell, agent, and test surfaces display the error without removing the last successful output or parsed failures.
- Clear errors after a successful read or task switch. Keep the existing final read, stale-result rejection, unmount cleanup, and non-overlapping polling behavior.
- Treat `ENOENT` before the first successful read as output not created yet. After a successful read, including an empty file, a missing file remains observable. Permission and I/O failures are always reported.
- Keep parsing and rendering in each surface. All three callers retain their existing byte limits and one-second intervals.

## Validation

- Root typecheck passes, including test-support types.
- All 1,018 workbench tests in 81 files pass with no failures or skips. Tests cover custom intervals, byte-limit changes, terminal transitions, final-read failures, recovery, task switches, stale promises, and unmount.
- Review regressions cover delayed file creation, empty successful reads, later file deletion, and initial permission/I/O failures. Both missing-initial-file cases fail on the first PR commit.
- All three surface error-display regressions fail against the unchanged runtime on `main` and pass with this change.
- Runtime build, package entrypoint and SDK mirror checks, and all six real PTY startup runs pass.
- GitNexus impact is LOW, with three direct hook callers and no indexed execution processes. The index refresh failure recorded in #2278 remains a tooling limitation; the diff is reviewed directly.
- Hosted test CI remains disabled at the repository owner's request. No hosted test workflow was enabled, dispatched, or rerun. Native Windows and macOS validation is unavailable on this Linux host.

Closes #1824.

---

> [!NOTE]
> **Low Risk**
> Workbench TUI output polling only; no auth, data, or core runtime behavior changes beyond clearer error display.
> 
> **Overview**
> **`useTaskTail`** now takes a named options object (`taskId`, `status`, `maxBytes`, **`pollIntervalMs`**) and returns **`{ content, error }`** instead of a bare string. Read failures surface an error message while **keeping the last successful tail**; errors clear on recovery or task switch. **`ENOENT`** before any successful read is treated as “output not created yet” (no UI error); after a read, a missing file is reported, and permission/I/O errors always are.
> 
> **Agent, Shell, and Test** workbench surfaces consume the new shape and show **“Output read failed: …”** without dropping parsed failures or prior output. Tests cover polling intervals, terminal status transitions, stale reads, and the ENOENT edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b816be92b1fd44d027049f39cc61c5729ee1bb13. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: runtime/src/tui/workbench/surfaces/useTaskTail.ts [hunk 1]
```diff
@@ -30,24 +40,33 @@ export function useTaskTail(
...
         // Retain the last successful tail after a transient disk failure.
-        if (active) logError(error);
+        if (active) {
+          setTail((current) => ({
+            taskId: id,
+            content: current.taskId === id ? current.content : "",
+            error: error instanceof Error ? error.message : String(error),
+          }));
+          logError(error);
+        }
```
[the remark this answers is not in the document]
Comment (7etsuo) [reply runtime/src/tui/workbench/surfaces/useTaskTail.ts:65]: Fixed in b816be92b. ENOENT stays hidden until the current task selection has a successful read. A later missing file remains visible as an error, with the last output retained. This also covers a previously empty file. EACCES, EPERM, and EIO remain visible from the first read.

Local validation passes all 1,018 workbench tests, typecheck, build, and six PTY startup runs. The two missing-initial-file regressions fail on the first PR commit.
